### PR TITLE
[fix](nereids) Fix bugs of group by constants 

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: close #xxx

## Problem summary

```sql
select 2 from nation group by 1
```

it should produce 2 would the table is not empty. Before this PR, the execution of nereids generated plan would produce empty result set

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.55 seconds
 load time: 631 seconds
 storage size: 17123722848 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221223130536_clickbench_pr_68036.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateGroupByConstant.java [hunk 1]
```diff
@@ -50,12 +50,18 @@ public Rule build() {
...
                 expression = FoldConstantRule.INSTANCE.rewrite(expression);
                 if (!(expression instanceof Literal)) {
                     slotGroupByExprs.add(expression);
+                } else {
+                    lit = expression;
                 }
             }
+            if (slotGroupByExprs.isEmpty() && lit != null) {
+                slotGroupByExprs.add(lit);
+            }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateGroupByConstant.java:64]: why we must keep at least one group expression?
if all group by expressions are constant, it is just like aggregate without group by keys. So in theory, we do not need to keep any constant group by key

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateGroupByConstant.java:64]: 和旧优化器保持一致，否则BE处理上会出问题

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateGroupByConstant.java:64]: add comment to explain why we need to keep at least one group expression~

**Review by morrySnow — APPROVED** [review]

**Pull request merged**